Sources/protoc-gen-swift/ExtensionGenerator.swift

```diff
@@ -19,6 +19,101 @@ import PluginLibrary
 import SwiftProtobuf
 
 class ExtensionGenerator {
+
+    // Helper to hold all the extensions when generating things across all
+    // the extensions within a given file.
+    class Set {
```

Comment:
Chatting offline, it might make sense to flip the model, make `ExtensionSet` the outer class and `ExtensionGenerator` the inner one.  That also can kill `register(extensions:)` and instead have the generators created by the `ExtensionSet` on demand.

But doing that will will be a pretty large churn on the ExtensionGenerator file, want me to do that in this PR (as an added commit) or do it as a followup PR?